core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -717,14 +717,18 @@ public Details newInstance(StaplerRequest req, JSONObject formData) throws FormE
                 if(!Util.fixNull(pwd).equals(Util.fixNull(pwd2)))
                     throw new FormException("Please confirm the password by typing it twice","user.password2");
 
+                User user = Util.getNearestAncestorOfTypeOrThrow(req, User.class);
+                UserProperty p = user.getProperty(Details.class);
```

Comment:
Did you check **all** the paths that lead to a User and ensured they all have the property set? Example: after changing security realm.

My concern is the `getProperty` could return `null` but you explicitly say you want a `@Nonnull` property and there is no check around that in your code.

Reply:
good point! updated in the latest commit.